openfecli/commands/gather.py

```diff
@@ -0,0 +1,95 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+import click
+from openfecli import OFECommandPlugin
+import pathlib
+from openfecli.utils import write
+
+
+def is_results_json(f):
+    # sanity check on files before we try and deserialize
+    return 'estimate' in open(f, 'r').read(20)
+
+
+def load_results(f):
+    # path to deserialized results
+    import json
+    from gufe.tokenization import JSON_HANDLER
+
+    return json.load(open(f, 'r'), cls=JSON_HANDLER.decoder)
+
+
+def get_names(result):
+    # Result to list of ligand names
+    return tuple(list(result['unit_results'].values())[0]['name'].split()[:2])
+
+
+def get_type(f):
+    if 'solvent' in f:
+        return 'solvent'
+    elif 'vacuum' in f:
+        return 'vacuum'
+    else:
+        return 'complex'
+
+@click.command(
+    'gather',
+    short_help="Gather DAG result jsons for entire network into single file"
+)
+@click.argument('rootdir',
+                type=click.Path(dir_okay=True, file_okay=False,
+                                         path_type=pathlib.Path),
+                required=True)
+@click.argument('output',
+                type=click.File(mode='w'),
+                required=True)
+def gather(rootdir, output):
+    """Gather DAG result jsons and write to single tsv file"""
+    from collections import defaultdict
+    import glob
+    import numpy as np
+
+    # 1) find all possible jsons
+    json_fns = glob.glob(str(rootdir) + '**/*json', recursive=True)
+
+    write(json_fns)
+
+    # 2) filter only result jsons
+    result_fns = filter(is_results_json, json_fns)
+
+    # 3) pair legs of simulations together
+    legs = defaultdict(dict)
+
+    for result_fn in result_fns:
+        result = load_results(result_fn)
+        if result is None:
+            continue
+
+        names = get_names(result)
+        simtype = get_type(result_fn)
+
+        legs[names][simtype] = result['estimate'], result['uncertainty']
+
+    # 4 for each ligand pair, write out the DDG
+    output.write('ligand pair\testimate (kcal/mol)\tuncertainty\n')
+    for ligpair, vals in legs.items():
+        if 'complex' in vals:
+            DG1_mag, DG1_unc = vals['complex']
+            DG2_mag, DG2_unc = vals['solvent']
+        else:
+            DG1_mag, DG1_unc = vals['solvent']
+            DG2_mag, DG2_unc = vals['vacuum']
+
+        DDG = DG1_mag - DG2_mag
```

Comment:
can both @RiesBen and @IAlibay check I've got my signs the right way round here

`vals['complex']` will be the DG of `L1 -> L2` in complex phase

Reply:
I might be getting muddled here, but isn't it:

hydration = vacuum - solvent
binding = solvent - complex

@RiesBen please double check

Reply:
no I'm the wrong way round

Reply:
aye :)